src/components/forms/SelectField/AsyncSelectField.js

```diff
@@ -1,29 +1,25 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import {useAsync} from 'react-use';
 
-import {SelectField} from './SelectField';
+import SelectField from './SelectField';
 
 const withOptions = WrappedComponent => {
   return ({optionsRetriever, ...props}) => {
     const {loading, error, value: options} = useAsync(optionsRetriever);
 
-    if (loading) {
-      return (
-        <WrappedComponent
-          options={[{label: 'Loading...', value: null}]} // Display loading message
-          {...props}
-        />
-      );
-    }
-
     if (error) {
-      return <div>Error loading options: {error.message}</div>; // Display error message
+      throw Error;
     }
 
-    return <WrappedComponent options={options} {...props} />;
+    return <WrappedComponent options={options} isLoading={loading} {...props} />;
   };
 };
 
 const AsyncSelectField = withOptions(SelectField);
-
+AsyncSelectField.propTypes =
+  SelectField.propTypes &
+  {
+    optionsRetriever: PropTypes.func.isRequired,
+  };
```

Comment:
I had to fix this in main for the Email component: d50e2f6ad2155e9bf0d77b24c08eab44e3233737 - I suspect that the `Foo.propTypes` get stripped out in production builds and it causes those storybook stories to crash.

You can define the common/shared props in a variable and use that to reduce the duplication, but getting them from `SelectField.propTypes` is not reliable.